backend/src/apiserver/resource/resource_manager.go

```diff
@@ -1124,7 +1124,7 @@ func (r *ResourceManager) ReadArtifact(runID string, nodeID string, artifactName
 	if run.WorkflowRuntimeManifest == "" {
 		return nil, util.NewInvalidInputError("read artifact from run with v2 IR spec is not supported")
 	}
-	execSpec, err := util.NewExecutionSpec([]byte(run.WorkflowRuntimeManifest))
```

Comment:
nice catch and this is the only place that calls `NewExecutionSpec()` which assumes the content is YAML. The content is in JSON format. sorry that I didn't catch the bug when doing the abstraction interface.

Reply:
/lgtm

Reply:
there is a `TestReadArtifact_Succeed` test case and it doesn't fail. I guess the testing data can pass the `yaml.Unmarshal` magically. thinking if we can update the test case to catch up with this kind of error. I can work on that.

Reply:
> which assumes the content is YAML. The content is in JSON format

I had the impression that because YAML is a superset of JSON, normally YAML serialization/deserialization libraries can handle both YAML and JSON (Didn't check if that's the case here, I probably got this impression from some Python library)

> there is a TestReadArtifact_Succeed test case and it doesn't fail.

I noticed the test as well. I think in my debugging of KFP runtime, `run.WorkflowRuntimeManifest` at this point doesn't carry the `TypeMeta` part. It was unclear to me that if it should at this point, or maybe the type needs to be told (by passing `util.ArgoWorkflow`) like the rest `NewExeuctionSpec*` calls. The latter seems making sense to me.

Reply:
yea, I would say it should carry `TypeMeta` and also specify `util.ArgoWorkflow`. at least so far we haven't supported dynamically switching orchestration engine yet :-p